src/main/java/io/bf2/kafka/authorizer/CustomAclAuthorizer.java

```diff
@@ -422,22 +423,17 @@ public List<CompletionStage<AclCreateResult>> createAcls(AuthorizableRequestCont
                 final CompletionStage<AclCreateResult> result;
 
                 if (!binding.entry().principal().startsWith(CustomAclBinding.USER_TYPE_PREFIX)) {
-                    /* Reject ACL operations as invalid where the principal named in the ACL binding is the principal performing the operation */
                     log.info("Rejected attempt by user {} to create ACL binding with invalid principal name: {}",
                             requestContext.principal().getName(),
                             binding.entry().principal());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL);
-                } else if (Objects.equals(toString(requestContext.principal()), binding.entry().principal())) {
-                    /* Reject ACL operations as invalid where the principal named in the ACL binding is the principal performing the operation */
-                    log.info("Rejected attempt by user {} to self-assign ACL binding",
-                            requestContext.principal().getName());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL);
-                } else if (hasPrincipalBindings(binding.entry().principal())) {
+                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_NON_USER_PRINCIPAL);
+                }
+                else if (hasPrincipalBindings(binding.entry().principal())) {
```

Comment:
nit: the position of the `else if` changes. 

I don't care one way or the other but I do value consistency.

Reply:
done, and updated my codestyle